### DISCUSSION TITLE
Parse short flag with no space before value

### DISCUSSION THREAD

**Discussion by wg** [pacak/bpaf#120] [Q&A] [answered]
The assertion at https://github.com/pacak/bpaf/blob/0.7.2/src/arg.rs#L99 fails when trying to parse an argument like `-Obits=2048`. Is there any way to make this work? OpenSSH tools including `ssh-keygen` support this argument style.

**Comment by pacak** [comment, marked as answer]
The value here being `bits=2048` and a short key `-O`?

**Reply by wg** [reply]
Yup exactly!

**Reply by pacak** [reply]
You can parse pretty much anything by hands by using a combination of `any`, `parse` and `anywhere`:

```
    let mystery = any::<String>("OUTPUT")
        .parse(|s| /* this parses -Obits=2048 into whatever you want or fails with some error message*/)
        .anywhere();
```

This will mess up the `--help` output so I'd slap `.hide()` on top of that and add an alternative that takes `-O`, contains the help message you want to see but fails and you combine those two with `construct!([mystery, help_n_fail])`.

Are you trying to implement some drop in replacement for `ssh-keygen`? If you don't need the value yourself and only want to pass it to `ssh-keygen` you can parse whatever you want and consume the rest with `any`.

**Reply by pacak** [reply]
https://github.com/pacak/bpaf/blob/master/examples/find.rs - some examples

**Reply by wg** [reply]
Thank you! I tried `any` but get the same error, it seems like that check is performed before the parsers are applied?

**Reply by pacak** [reply]
Hmm... You are right, this check needs to succeed earlier. Let me try something...

**Reply by pacak** [reply]
Can you explain your use case a bit more?

**Reply by wg** [reply]
Sure, I'm making a code signing tool that needs to be compatible with `ssh-keygen` so it can be used to sign git commits. Unfortunately git is hardcoded to pass `-Ooption=value` instead of the more reasonable `-O option=value` which is also supported by ssh-keygen.

**Reply by pacak** [reply]
I'll make a change to get it working - probably tomorrow.

**Reply by pacak** [reply]
If you use `master` branch - it should parse it as expected with just short option `O`, see code here: https://github.com/pacak/bpaf/pull/122

Also thank you for an interesting case.

**Reply by wg** [reply]
Thank you for getting this working so quickly! It's rare running a fellow Rust programmer here in JST =) And thank you for bpaf! I'm really enjoying the elegant API and the clean separation between arguments/commands and the data structures they're used to build.

**Reply by pacak** [reply]
> It's rare running a fellow Rust programmer here in JST =)

Even more rare than you think because I'm not :)

> I'm really enjoying the elegant API and the clean separation between arguments/commands and the data structures they're used to build.

Category theory, Functors, Applicatives, Monads - all those things are your friends.

**Comment by pacak** [comment]
Released 0.7.3 with this fix included.
